tests/test_reply_handling.py

```diff
@@ -502,6 +505,25 @@ def get_response(client, x):
     assert response_bare.__class__ is suds.sax.text.Text
     assert response_bare == data
 
+    inject = dict(reply=suds.byte_str("""\
+<?xml version="1.0"?>
+<env:Envelope xmlns:env="http://www.w3.org/2003/05/soap-envelope">
+  <env:Body>
+    <env:Fault>
+      <faultcode>env:Client</faultcode>
+      <faultstring>Dummy error.</faultstring>
+    </env:Fault>
+  </env:Body>
+</env:Envelope>
+"""), status=http.client.INTERNAL_SERVER_ERROR)
+    e = pytest.raises(suds.WebFault, client_bare.service.f, __inject=inject).value
+    try:
+        _test_fault(e.fault, False)
+        assert e.document.__class__ is suds.sax.document.Document
+        assert str(e) == "Server raised fault: 'Dummy error.'"
+    finally:
+        del e  # explicitly break circular reference chain in Python 3

```

Comment:
Can you explain what this is doing? What is there a circular reference to?

Reply:
I have referred to other test codes, and all exceptions are explicitly deleted. I think the main reason is to avoid exception objects being circularly referenced and not garbage collected.